src/averageAges.js

```diff
@@ -54,8 +68,20 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+ function calculateAverageAgeDiff(people, onlyWithSon) {
+  const family = onlyWithSon
+
+    ? people.filter(person =>
+      people.some(mother => mother.name === person.mother)
+    && person.sex === 'm')
+
+    : people.filter(person =>
+      people.some(mother => mother.name === person.mother));
+
+  const differenceInAge = family.map(person =>
+    person.born - people.find(mother => mother.name === person.mother).born);
```

Comment:
In some cases, more variables it's good thing)
For example, in this function you can create separate variable for mother, and then use this variable in age calculation
It'll help to better code understanding)